tar/read.c

```diff
@@ -148,10 +148,17 @@ read_archive(struct bsdtar *bsdtar, char mode, struct archive *writer)
 	int			  r;
 
 	while (*bsdtar->argv) {
-		if (archive_match_include_pattern(bsdtar->matching,
-		    *bsdtar->argv) != ARCHIVE_OK)
-			lafe_errc(1, 0, "Error inclusion pattern: %s",
-			    archive_error_string(bsdtar->matching));
+		if ((*bsdtar->argv)[0] == '-' && (*bsdtar->argv)[1] != '\0' && strcmp(*bsdtar->argv, "-") != 0) {
```

Comment:
I'm not sure this is the right way to fix this.  Could we split it out and discuss it as a separate PR?